utilities/translation/tfcomp.js

```diff
@@ -1,105 +1,182 @@
+// Detect missing translation items between translation files
+// USAGE:  node tfcomp.js path1 path2
+// Where path1 and path2 : path to a folder containing json files of translation
 
 'use strict';
 
 const path = require('path');
+const fs = require('fs');
 
-// get the filenames
-const jsonA = process.argv[2];
-const jsonB = process.argv[3];
+// get the files directory
+const pathEn = process.argv[2];
+const pathFr = process.argv[3];
 
-console.log('Comparing', jsonA, jsonB);
 
-// load JSON files
-const A = require(path.join(process.cwd(), jsonA));
-const B = require(path.join(process.cwd(), jsonB));
+const EN_PATH = path.resolve(process.cwd(), pathEn);
+const FR_PATH = path.resolve(process.cwd(), pathFr);
+
+const enJsonNames = fs.readdirSync(EN_PATH);
+const frJsonNames = fs.readdirSync(FR_PATH);
 
 // cache the object type
 const object = typeof ({});
 
+let errMsg = '';
+
 // Arrays to save differences in
-let miss1 = [];
-let miss2 = [];
+let enMissList = null;
+let frMissList = null;
+let enFileMissList = [];
+let frFileMissList = [];
+
+const jsonFiles = buildJsonFileArray();
+
+jsonFiles.forEach(function (jsonFile) {
+
+  // Arrays to save differences in
+  enMissList = [];
+  frMissList = [];
+
+  if(jsonFile.en && jsonFile.fr){
+
+    // load JSON files
+    let enTranslateObject = require(jsonFile.en);
+    let frTranslateObject = require(jsonFile.fr);
+
+    checkSubDict(enTranslateObject, frTranslateObject, '');
+  }else{
+    //add to the missed files list
+    !jsonFile.en ? enFileMissList.push(jsonFile.fr) : frFileMissList.push(jsonFile.en);
+  }
+
+
+  // Report items in french translation but missing from english translation
+  if (enMissList.length > 0) {
+    errMsg += '\nMissing from ' + jsonFile.en + ': \n';
+    enMIssList.sort();
+    errMsg += enMissList.join('\n');
+    errMsg += '\n\n';
+  }
+
+  // Report items in filename1 but missing from filename2
+  if (frMissList.length > 0) {
+    errMsg += 'Missing from ' + jsonFile.fr + ': \n';
+    frMissList.sort();
+    errMsg += frMissList.join('\n');
+    errMsg += '\n\n';
+  }
+});
+
+if(enFileMissList.length > 0){
+  errMsg += '\n Missing english correspondent file for : \n';
+  enFileMissList.sort();
+  errMsg += enFileMissList.join('\n');
+  errMsg += '\n\n';
+}
+
+if(frFileMissList.length > 0){
+  errMsg += '\n Missing french correspondent file for : \n';
+  frFileMissList.sort();
+  errMsg += frFileMissList.join('\n');
+  errMsg += '\n\n';
+}
+
+if (errMsg) {
+  console.error(errMsg);
+}
+
+function buildJsonFileArray (){
+  let jsonList = [];
+
+  enJsonNames.forEach(function (enJsonName) {
+    const ind = frJsonNames.indexOf(enJsonName);
+    let item = {
+      en : path.resolve(EN_PATH, enJsonName),
+      fr : null
+    };
+
+    if(ind >= 0) {
+      item.fr = path.resolve(FR_PATH, frJsonNames[ind]);
+    }
+
+    jsonList.push(item);
+  });
+
+  const missedFromEnJsonNames = frJsonNames.filter(function (frJsonName) {
```

Comment:
For future reference, you can use a new JavaScript array method to simplify this code: [Array.prototype.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes).  It would look like this:

```js
const missingFromEnJsonNames = frJsonNames.filter(frJsonName => !enJsonNames.includes(frJsonName));
```